.ci/pipelines/build.Jenkinsfile

```diff
@@ -115,15 +115,6 @@ pipeline {
                     ],
                     wait: false
 
-                // schedule another job for all the older 4.x OCP versions which runs only on Fridays (via when directive in job)
-                build job: 'cloud-on-k8s-e2e-tests-ocp-all-but-latest',
-                    parameters: [
-                        string(name: 'JKS_PARAM_OPERATOR_IMAGE', value: operatorImage),
-                        string(name: 'branch_specifier', value: GIT_COMMIT)
-                    ],
-                    quietPeriod: 86400, // add a 24 hour delay to this job to run it over the weekend
-                    wait: false
-
```

Comment:
Another approach I can think of to not introduce a new `extended-e2e` job and avoid duplicating build/push/upload steps would be to start the nightly jobs conditionally here in `build.Jenkinsfile `, and to trigger the `cloud-on-k8s-nightly-build` job Monday to Saturday:
```groovy
 if isWeekDay() {
  build job: 'cloud-on-k8s-e2e-tests-stack-versions', ...
  build job: 'cloud-on-k8s-e2e-tests-gke-k8s-versions', ...
  build job: 'cloud-on-k8s-e2e-tests-aks', ...
  build job: 'cloud-on-k8s-e2e-tests-kind-k8s-versions', ...
  build job: 'cloud-on-k8s-e2e-tests-ocp', ...
  build job: 'cloud-on-k8s-e2e-tests-ocp-all-but-latest', ...
  build job: 'cloud-on-k8s-e2e-tests-eks', ...
  build job: 'cloud-on-k8s-e2e-tests-eks-arm', ...
  build job: 'cloud-on-k8s-e2e-tests-resilience', ...
  build job: 'cloud-on-k8s-e2e-tests-ocp-all-but-latest', ...
  build job: 'cloud-on-k8s-e2e-tests-tanzu', ...
} else {
   build job: 'cloud-on-k8s-e2e-tests-ocp-all-but-latest', ...
   build job: 'cloud-on-k8s-e2e-tests-tanzu, ...
}
```

Reply:
Hm yeah, maybe that is better. I went with a separate Jenkinsfile to minimise logic inside the pipeline. But as you said that comes at the cost of duplicating some steps.